Tech magma

sangallidavide opened the pull request:
Interface to the magma library for diagonalization of the BSE (also on GPU)

Developments by @andrea-ferretti and @blmelp 

Pull request opened after suggesion of @andrea-ferretti

nicspalla committed:
added configure support for magma libraries. support for internal library compilation not yet available.

andrea-ferretti committed:
Merge branch 'develop' into tech/magma

Conflicts:
	configure

andrea-ferretti committed:
cuda compilation fixed

andrea-ferretti committed:
Revert "cuda compilation fixed"

This reverts commit ffc7c5e0042c795848745d86c3ee1645b4087df9.

andrea-ferretti committed:
Version 5.1.0,  Revision 22124,  Hash 407f3a474

    MODIFIED *  include/driver/version.h dipoles/DIPOLE_overlaps.F

    NEW *

    Bugs:
    - workaround for issue #729 to fix compilation

    Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

andrea-ferretti committed:
tentative implementation of interface to magma_Xgeev_m (X=z,c)

andrea-ferretti committed:
compilation without magma fixed

andrea-ferretti committed:
thread safeness added to magma interfaces

blmelp committed:
Magma implementation for diagonalization (overwrites serial implementation)

blmelp committed:
Changed magma functions to multiple gpu version (with 2 gpus hardcoded)

andrea-ferretti committed:
Merge remote-tracking branch 'yambo-devel/tech/magma' into tech/magma

Conflicts:
	config/report.in
	include/driver/version.h
	src/modules/.objects

andrea-ferretti committed:
tentaitve fix to magma interfaces (see comments by blmelp,
Blanca Mellado Pinto)

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Version 5.2.0,  Revision 23239,  Hash b626a0b2cb

MODIFIED *  configure include/version/version.m4

DELETED *  include/driver/version.h

Changes:
- deleted old version.h file (not needed anymore)

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide commented:
There is also a discussion in this connected pull-request: https://github.com/blmelp/yambo/pull/1

Was this solved?

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Version 5.2.0,  Revision 23256,  Hash b06be9e3ad

MODIFIED *  configure include/version/version.m4

Changes:
- [yambo]: configure re-generated after merge

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Version 5.2.0,  Revision 23297,  Hash 8a9239ac3d

MODIFIED *  config/m4/acx_report.m4 config/report.in configure include/version/version.m4

Bugs:
- Few small fixes in the configure

Additions:
-

Changes:
-

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide commented:
> There is also a discussion in this connected pull-request: [blmelp#1](https://github.com/blmelp/yambo/pull/1)
> 
> Was this solved?

@andrea-ferretti  should we proceed with the merge?

I checked the code. I see:
- the magma can be linked only externally for now
- the magma are used only in case of serial diagonalization with complex eigenvalues (in practice bse with coupling). Is this correct?

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Version 5.2.0,  Revision 23307,  Hash 7175ddda6a

MODIFIED *  include/version/version.m4 modules/mod_linear_algebra.F

Changes:
- [yambo] Added few messages

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Version 5.2.0,  Revision 23308,  Hash 2e17f6a970

MODIFIED *  config/m4/acx_report.m4 config/m4/magma.m4 config/mk/global/actions/compile_external_libraries.mk config/mk/global/actions/download_external_libraries.mk configure include/version/version.m4 lib/archive/Makefile.loc lib/archive/package.list

NEW *  lib/magma/Makefile.loc

Changes:
- [yambo] few more steps to prepare internal magma compilation

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

andreamarini commented:
Dear Davide, MAGMA libs are necessary to use this branch? 

If the answer is yes a documentation is needed in order to install them properly.

This is also needed in order to add the lib to the robots libraries. 

THX

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/maintenance-master'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Merge remote-tracking branch 'blmelp/valencia_magma' into tech-magma

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Version 5.2.0,  Revision 23323,  Hash 57187815cf

MODIFIED *  configure include/version/version.m4

DELETED *  config/stamps_and_lists/project_dependencies.stamp

Changes:
- [yambo] Configure re-generated after merge

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Version 5.2.0,  Revision 23469,  Hash fde0c4f549

MODIFIED *  configure include/version/version.m4

DELETED *  config/stamps_and_lists/project_dependencies.stamp

Changes:
- configure re-generated

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Version 5.2.0,  Revision 23470,  Hash cbf592b6b7

MODIFIED *  config/m4/magma.m4 configure include/version/version.m4 lib/magma/Makefile.loc

NEW *  lib/magma/MGmake.inc_lib

DELETED *  config/stamps_and_lists/project_dependencies.stamp

Changes:
- [magma]  created first version of magma Makefile / make.inc

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Version 5.2.0,  Revision 23471,  Hash 27b3538f80

MODIFIED *  config/m4/acx_report.m4 config/m4/magma.m4 config/mk/global/actions/compile_internal_libraries.mk config/mk/global/libraries.mk config/report.in configure include/version/version.m4 modules/.objects

NEW *  lib/magma_fmodules/.objects

RENAMED *  modules/mod_magma2.F -> lib/magma_fmodules/mod_magma2.F modules/mod_magma2_cfortran.F -> lib/magma_fmodules/mod_magma2_cfortran.F modules/mod_magma2_common.F -> lib/magma_fmodules/mod_magma2_common.F modules/mod_magma2_dfortran.F -> lib/magma_fmodules/mod_magma2_dfortran.F modules/mod_magma2_sfortran.F -> lib/magma_fmodules/mod_magma2_sfortran.F modules/mod_magma2_zfortran.F -> lib/magma_fmodules/mod_magma2_zfortran.F

Changes:
- [magma] fmodules coved inside lib/magma_fmodules folder as discussed
  during last meeting

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Version 5.2.0,  Revision 23472,  Hash 5485636137

MODIFIED *  config/m4/magma.m4 config/mk/global/defs.mk.in config/mk/global/libraries.mk config/report.in configure include/version/version.m4 lib/magma_fmodules/.objects

Changes:
- [magma_fmodules] Check for modules refined

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-gpu'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-gpu'

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-master'

sangallidavide committed:
Version 5.3.0,  Revision 23811,  Hash 5e711996fd

MODIFIED *  configure include/version/version.m4

Changes:
- configure re-generated

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/tech-ydriver'

sangallidavide committed:
Merge branch 'tech-master' into tech-magma

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/5.3'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/5.3'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/5.3'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/5.3'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/5.3'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/master'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/master'

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/master'

sangallidavide committed:
Version 5.3.0,  Revision 23925,  Hash 5e1a3aea10

MODIFIED *  configure include/version/version.m4 ypp/excitons/excitons_bands.F ypp/excitons/excitons_interpolate_setup.F ypp/plotting/plot_interpolated_values.F

Changes:
- Built in bands available also in excitons interpolation

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge branch 'master' into tech-magma

sangallidavide commented:
> Dear Davide, MAGMA libs are necessary to use this branch?

No, they are optional. I'm not the expert here. @andrea-ferretti did most of the coding

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/master'

sangallidavide committed:
Version 5.3.0,  Revision 23939,  Hash 2d982002ae

MODIFIED *  configure include/version/version.m4 dipoles/Berry_polarization_EQ.F dipoles/DIPOLE_covariant.F hamiltonian/Build_W_operator.F nloptics/NL_Berry_current.F nloptics/NL_current.F

Bugs:
- List of changes propsed in https://github.com/yambo-code/yambo/issues/172
  Branch created after quick discussion with Claudio and Myrta

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

andrea-ferretti committed:
Merge branch '5.3' into tech-magma

sangallidavide committed:
Merge branch 'master' into tech-magma

sangallidavide committed:
Version 5.3.0,  Revision 23949,  Hash cdf88af1ad

MODIFIED *  configure include/version/version.m4

NEW *  Discussion_AD_DS_20250225.dat

Additions:
- Created file with discussion with AndreaF

Patch sent by:  Davide Sangalli <davide.sangalli@gmail.com>

sangallidavide committed:
Merge branch '5.3' into tech-magma

andrea-ferretti committed:
one more step for magma support impl

andrea-ferretti committed:
Merge branch 'tech-magma' of github.com:yambo-code/yambo into tech-magma

Conflicts:
	configure

andrea-ferretti committed:
compilation fixed

andrea-ferretti committed:
GEEV interface removed and included in SERIAL_diagonalization

andrea-ferretti committed:
HEEV also eliminated from module and merged in SERIAL_HERM_diago.

andrea-ferretti committed:
further rationalization of LA drivers

andrea-ferretti committed:
LA interfaces and modules further rationalized

andrea-ferretti committed:
cleanup of interfaces in the case of magma

andrea-ferretti committed:
more cleanup

nicspalla committed:
added flag to prevent lapack to install in lib64 directory: see issue #189

andrea-ferretti committed:
Merge branch 'tech-master' into tech-magma

Conflicts:
	configure

andrea-ferretti committed:
Merge branch 'tech-magma' of github.com:yambo-code/yambo into tech-magma

andrea-ferretti committed:
Merge branch 'tech-magma' of github.com:yambo-code/yambo into tech-magma

andrea-ferretti committed:
cleanup

sangallidavide committed:
Merge remote-tracking branch 'yambo-gpl/master' into fixes-issues-172

sangallidavide committed:
Merge pull request #176 from sangallidavide/fixes-issues-172

Fixes issues 172

andrea-ferretti committed:
magma configuration fixed; compilation of internal fmodules working

andrea-ferretti committed:
nternal compilation of magma implemented

sangallidavide committed:
Merge branch 'master' into tech-magma
